File: backend/back-pasteleria/src/main/java/cl/milsabores/pasteleria/dto/CheckoutRequest.java
package cl.milsabores.pasteleria.dto;

import jakarta.validation.Valid;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import lombok.Data;

import java.util.List;

@Data
public class CheckoutRequest {

    @NotEmpty
    private List<@Valid CartItemRequest> items;

    @NotBlank
    private String firstName;

    @NotBlank
    private String lastName;

    @Email
    @NotBlank
    private String email;

    @NotBlank
    private String phone;

    private String run;
    private String birthDate;

    @NotBlank
    private String deliveryOption;

    private String street;
    private String region;
    private String comuna;
    private String address;

    private String branch;
    private String pickupDate;
    private String pickupTimeSlot;

    @NotBlank
    private String paymentMethod;

    private String notes;
    private String couponCode;
}

File: app/src/main/java/com/example/pasteleriamilssaboresandroid/data/network/dto/CheckoutRequest.kt
package com.example.pasteleriamilssaboresandroid.data.network.dto

import kotlinx.serialization.Serializable

@Serializable
data class CheckoutRequest(
    val items: List<CartItemRequest>,
    val firstName: String,
    val lastName: String,
    val email: String,
    val phone: String,
    val run: String? = null,
    val birthDate: String? = null,
    val deliveryOption: String,
    val street: String? = null,
    val region: String? = null,
    val comuna: String? = null,
    val address: String? = null,
    val branch: String? = null,
    val pickupDate: String? = null,
    val pickupTimeSlot: String? = null,
    val paymentMethod: String,
    val notes: String? = null,
    val couponCode: String? = null
)

